src/vorta/views/archive_tab.py

```diff
@@ -297,8 +320,71 @@ def receive():
                         else:
                             self._set_status(params['message'])
 
-                dialog = choose_file_dialog(self, "Choose Extraction Point")
+                dialog = choose_file_dialog(self, "Choose Extraction Point", want_folder=True)
                 dialog.open(receive)
 
     def extract_archive_result(self, result):
         self._toggle_all_buttons(True)
+
+    def update_mount_button_text(self):
+        snapshot_name = self.selected_archive_name()
+        if not snapshot_name:
+            return
+
+        if snapshot_name in self.mount_points:
+            self.set_mount_button_mode('Unmount')
+        else:
+            self.set_mount_button_mode('Mount')
```

Comment:
mode = 'Unmount' if snapshot_name in self.mount_points else 'Mount'
self.set_mount_button_mode(mode)

Reply:
ok